sensor/common/scan/scan.go

```diff
@@ -163,57 +131,171 @@ func (s *LocalScan) enrichLocalImageFromRegistry(ctx context.Context, centralCli
 	}
 	defer s.scanSemaphore.Release(1)
 
-	log.Debugf("Enriching image locally %q numRegs %v", ci.GetName().GetFullName(), len(registries))
-
-	if len(registries) == 0 {
-		// no registries provided, try with no auth.
-		reg, err := s.createNoAuthImageRegistry(ctx, ci.GetName(), s.regFactory)
-		if err != nil {
-			return nil, errors.Join(pkgErrors.Wrapf(err, "unable to create no auth registry for %q", ci.GetName()), ErrEnrichNotStarted)
-		}
-		registries = append(registries, reg)
-	}
+	log.Debugf("Enriching image locally %q, namespace %q, requestID %q, force %v", srcImage.GetName().GetFullName(), namespace, requestID, force)
 
 	errorList := errorhelpers.NewErrorList("image enrichment")
 
-	image := types.ToImage(ci)
-	image.Notes = make([]storage.Image_Note, 0)
-
 	// Enrich image with metadata from one of registries.
-	reg := s.enrichImageWithMetadata(errorList, registries, image)
+	reg, pullSourceImage := s.getImageWithMetadata(ctx, errorList, namespace, srcImage)
 
 	// Perform partial scan (image analysis / identify components) via local scanner.
-	scannerResp := s.fetchImageAnalysis(ctx, errorList, reg, image)
+	scannerResp := s.fetchImageAnalysis(ctx, errorList, reg, pullSourceImage)
 
 	// Fetch signatures associated with image from registry.
-	sigs := s.fetchSignatures(ctx, errorList, reg, image)
+	sigs := s.fetchSignatures(ctx, errorList, reg, pullSourceImage)
 
 	// Send local enriched data to central to receive a fully enrich image. This includes image vulnerabilities and
 	// signature verification results.
 	centralResp, err := centralClient.EnrichLocalImageInternal(ctx, &v1.EnrichLocalImageInternalRequest{
-		ImageId:        image.GetId(),
-		ImageName:      image.GetName(),
-		Metadata:       image.GetMetadata(),
+		ImageId:        srcImage.GetId(),
+		ImageName:      srcImage.GetName(),
+		Metadata:       pullSourceImage.GetMetadata(),
 		Components:     scannerResp.GetComponents(),
 		Notes:          scannerResp.GetNotes(),
 		ImageSignature: &storage.ImageSignature{Signatures: sigs},
-		ImageNotes:     image.GetNotes(),
+		ImageNotes:     pullSourceImage.GetNotes(),
 		Error:          errorList.String(),
 		RequestId:      requestID,
 		Force:          force,
 	})
 	if err != nil {
-		log.Debugf("Unable to enrich image %q: %v", image.GetName().GetFullName(), err)
-		return nil, pkgErrors.Wrapf(err, "enriching image %q via central", image.GetName())
+		log.Debugf("Unable to enrich image %q: %v", srcImage.GetName().GetFullName(), err)
+		return nil, pkgErrors.Wrapf(err, "enriching image %q via central", srcImage.GetName())
 	}
 
 	if errorList.Empty() {
-		log.Debugf("Retrieved image enrichment results for %q with id %q", image.GetName().GetFullName(), image.GetId())
+		log.Debugf("Retrieved image enrichment results from Central for %q with id %q (%v) components", srcImage.GetName().GetFullName(), srcImage.GetId(), centralResp.GetImage().GetComponents())
```

Comment:
`GetComponents()` at the image level is a count - see [here](https://github.com/stackrox/stackrox/blob/868de0637884bfc8f09c93eecb81226810a686b2/generated/storage/image.pb.go#L404). So in this case don't believe it will be noisy.